### ISSUE TITLE
More advanced DevOps

### ISSUE THREAD

**Issue by HeshamMeneisi** [simple-icons/simple-icons#2984]
### Kind of issue 
  - [x] Improvement
  - [ ] Bug
  - [ ] Other, namely:

### This issue concerns 
  - [x] The NPM Package
  - [x] The Packagist Package

#### 1. Husky Hooks

This is an improvement for contributors. If the package contains a pre-commit hook (using something like husky) that runs basic checks like linting and maybe a pre-push hook with more time-consuming operations like `npm test` and/or `npm run build`, it would save some time as simpler and rather straightforward issues would be discovered locally instead of waiting for Travis to fail.

I experienced this during my recent PR where Travis failed because the icon names weren't in correct alphabetical order (`npm run our-lint` failed). I know this was mentioned in the contribution guide but using hooks like that is more convenient and a common practice I have seen in most projects I worked on or contributed to. I can do this when I have some time if you agree.

#### 2. Semantic Release

On a different subject, I can also help setup semantic release instead of the current bi-monthly release if you are interested. So a release would be triggered by commits like:
`fix(github): update icon hex`
`feat(icons): add google icon`

Ideally, this would be done end to end using the semantic-release action. But we can leverage the current release action since a lot of work went into it by letting semantic-release decide whether or not this is a release then skip/run the release action based on this. Something like:

```yaml
steps:
  - name: Checkout
    uses: actions/checkout@v2
  - name: Semantic Release
    uses: cycjimmy/semantic-release-action@v2
    env:
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      
  - name: Release
    if: steps.semantic.outputs.new_release_published == 'true'
    uses: simple-icons/release-action@master
      with:
        repo-token: "${{ secrets.GITHUB_TOKEN }}"
        release-version: "${{ steps.semantic.outputs.new_release_version }}"
        release-notes: "${{ steps.semantic.outputs.new_release_notes }}"
```

This also enables the addition of other semantic-release plugins (automated changelog, docs, etc)

**Comment by ericcornelissen** [comment]
Hi @HeshamMeneisi, thanks for reaching out and giving some nice suggestions.

Up front I want to share with you our philosophy/goal for contributing this project, which should clarify my response. As contributing an icon to this project is pretty simple from a software development perspective, we aim to make the contributing process not too difficult/convoluted so that people that have little experience with software development, git, or GitHub can contribute without being scared away. :slightly_smiling_face: 

With that said, here is my response to your suggestions:

#### 1. Husky Hooks

I love Husky and I use it in most of my (WebDev) projects as well. The reason I haven't attempted to add it to this project yet is to avoid confusing contributors that are new to git an git hooks that get confused that they can't create a commit if the hook fails.

That said, I think this is a perfect opportunity to discuss adding Husky to this project! I suspect that most contributors that are going to run `npm install` when contributing to this project will actually understand what Husky is doing. And for those that don't we could explain it in [the Contributing Guidelines](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md).

So, with that, I would like to ask the @simple-icons/maintainers what they think about setting up Husky.


#### 2. Semantic Release

We are actually currently working on automating the release process. [The last release](https://github.com/simple-icons/simple-icons/pull/2982) was automatically created by [a custom GitHub Action](https://github.com/simple-icons/release-action). :tada: 

The reason we choose that over using semantic releases is, again, to keep contributing as simple as possible. For someone that is new to git and software development, having to understand semantic commit messages is a hurdle to get over. And if they do get over it in the first place, they may be afraid of doing it incorrectly.

Luckily, we are in the fortunate position that our changes notes can easily be generated based on the changes in a PR.

So in short, I'm rejecting the idea of using semantic releases in this project.

**Comment by runxel** [comment]
Hello @HeshamMeneisi,
thanks for the improvement suggestions!
Eric already laid out the most important points, and while the idea of semantic releases is already out of question, I'd say the same arguments can be used against Husky and therelike.

We really try to lower the hurdles for contributing (that's why we still link to SVGOMG e.g.) and I doubt many of the contributers actually use npm in the first place (based on the fact most people _still use SVGOMG_...).
I just fail to see any benefits when using pre-commit hooks here, just adding a layer of "complexity". Still, I can understand the reasoning for its use :)

Stay safe! 👋

**Comment by HeshamMeneisi** [comment]
Hey guys. I understand your approach to reducing complexity.

However, I need to clarify that my recommendation for semantic release was based on the merge commit (by maintainers) not the contributor commits, and to complement not replace the current release action.

When a maintainer merges something with feat or fix, it would trigger the release action (the one used right now) instead of having to wait for the scheduled release.

This is the addition to the current release action if it's triggered on [push] and uses semantic release to decide if this is a release.

```yaml
...
- name: Release
    if: steps.semantic.outputs.new_release_published == 'true'
    uses: simple-icons/release-action@master
...
```

With that said. It's sure up to you whether or not you'd want to do this (maybe you intentionally want a scheduled release).

**Comment by ericcornelissen** [comment]
That is a nice recommendation that I had never considered. However, I personally prefer sticking with releases on a regular basis. Although I do understand that people may want an SVG quicker than 1 or 2 weeks, I don't think we should create a release for every individual logo that is added.

Also, considering that the current approach is completely tool-agnostic, I prefer it over maintainers having to manually change the merge commit title to the semantic format. If we ever want to release something weird that is not detected by our custom Action, we can just add it in manually. I think that is less work overall for our case.

But again, thanks for the recommendation and especially thanks for this last comment, I never thought of that!

* * *

With that, I would like some input from other @simple-icons/maintainers no setting up huskye.

**Comment by ericcornelissen** [comment]
Given the lack of activity in this issue I'm going to close this for now. If anyone wants to bring up using Husky again feel free to respond in this thread or open a new issue :slightly_smiling_face:
